fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -2451,4 +2451,43 @@ private void checkSplitIndexBuildRange(int minSplit, int maxSplit,
             commit(context);
         }
     }
+
+    @Test
+    public void testMockedLocalityUtil() throws Exception {
+        final FDBDatabaseFactory factory = FDBDatabaseFactory.instance();
+        factory.setLocalityProvider(MockedLocalityUtil.instance());
+        FDBDatabase database = FDBDatabaseFactory.instance().getDatabase();
+
+        try (FDBRecordContext context = database.openContext()) {
+            openSimpleRecordStore(context, TEST_SPLIT_HOOK);
+            ArrayList<byte[]> keys = new ArrayList<>();
+            for (int i = 0; i < 50; i++) {
+                keys.add(recordStore.recordsSubspace().pack(i));
+            }
+
+            MockedLocalityUtil.init(keys, 0);
+            assertEquals(0, getRangeCount(MockedLocalityUtil.instance().getBoundaryKeys(context.ensureActive(), keys.get(0), keys.get(keys.size() - 1))));
```

Comment:
Maybe this can be pulled out.
```java
testRangeCount(0);
testRangeCount(1);
...
private testRangeCount(int rangeCount) {
    MockedLocalityUtil.init(keys, rangeCount);
    ... boudnaryKeys = MockedLocalityUtil.instance()
        .getBoundaryKeys(context.ensureActive(), keys.get(0), keys.get(keys.size() - 1));
    assertEquals(rangeCount, getRangeCount(boudnaryKeys));
}
```

Reply:
good idea, done.